pkg/snowflake/parser_test.go

```diff
@@ -137,6 +137,69 @@ from bar;`
 	}
 }
 
+func TestViewSelectStatementExtractor_ExtractDynamicTable(t *testing.T) {
+	basic := "create dynamic table foo lag = 'DOWNSTREAM' refresh_mode = 'AUTO' initialize = 'ON_CREATE' warehouse = COMPUTE_WH as select * from bar;"
+	caps := "CREATE DYNAMIC TABLE FOO LAG = 'DOWNSTREAM' REFRESH_MODE = 'AUTO' INITIALIZE = 'ON_CREATE' WAREHOUSE = COMPUTE_WH AS SELECT * FROM BAR;"
+	parens := "create dynamic table foo lag = 'DOWNSTREAM' refresh_mode = 'AUTO' initialize = 'ON_CREATE' warehouse = COMPUTE_WH as (select * from bar);"
+	multiline := `
+create dynamic table foo
+lag = 'DOWNSTREAM'
+refresh_mode = 'AUTO'
+initialize = 'ON_CREATE'
+warehouse = COMPUTE_WH
+as select * from bar;`
+
+	multilineComment := `
+create dynamic table foo
+lag = 'DOWNSTREAM'
+refresh_mode = 'AUTO'
+initialize = 'ON_CREATE'
+warehouse = COMPUTE_WH
+as
+-- comment
+select *
+from bar;`
+
+	comment := `create dynamic table foo comment = 'asdf' lag = 'DOWNSTREAM' refresh_mode = 'AUTO' initialize = 'ON_CREATE' warehouse = COMPUTE_WH as select * from bar;`
```

Comment:
In every example with `comment` it is placed in the "wrong" place. If you check `createDynamicTableOptions` you can see that the `Comment` is the last thing before the `query`. Please change it here and in the `ExtractDynamicTable()`.

Reply:
Thanks for this info. This appears to an inconsistency in what Snowflake does internally with the query, as fetching the query using `SHOW DYNAMIC TABLES LIKE foo` results in a query where the comment comes before all other parameters.

This could be upstream instability due to dynamic tables being a preview feature. I will make the parser handle both cases.

I totally get your hesitation with this change given the state of the framework in general, but I am also prepared to use this as a temporary bandaid that at least unblocks us and others with an urgent need to rely on Terraform to manage dynamic tables.